torch/nn/parallel/data_parallel.py

```diff
@@ -54,6 +54,13 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
 
         if device_ids is None:
             device_ids = list(range(torch.cuda.device_count()))
+
+        devices_memory = [torch.cuda.max_memory_allocated(i) for i in device_ids]
+        if max(devices_memory) / min(devices_memory) >= 3.0:
```

Comment:
this is great, but I think we should universally warn on such detection. Can you move this to torch/cuda/__init__.py where the other warnings are about old GPUs

Reply:
I'd gladly do so, but `max_memory_allocated()` returns 0 on my machine for all GPUs. The name of the method is misleading.

Reply:
what you want is not `max_memory_allocated`, you want the total device memory.
It looks like we didn't expose this in pytorch yet, let me send a PR in a few minutes

Reply:
@lemairecarl see https://github.com/pytorch/pytorch/pull/5386

Reply:
awesome, thanks!

@apaszke suggested in #5386 that we expose the whole struct. Have all info, should we also check the number of cuda cores? The GT710 has 192 cores, while the TITAN X has 3584 (almost 20x).